tests/sparseml/pytorch/sparsification/pruning/test_mfac_inverse.py

```diff
@@ -0,0 +1,92 @@
+# Copyright (c) 2021 - present / Neuralmagic, Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+
+import pytest
+import torch
+
+from flaky import flaky
+from sparseml.pytorch.sparsification.pruning import (
+    FisherInverseFast,
+    FisherInverseFastBlock,
+    FisherInverseFastSmallBlocks,
+)
+
+
+# precent-wise precision in terms of the mean of the resulting tensors
+PRECISION = 0.00001
+
+
+@pytest.mark.skipif(
+    os.getenv("NM_ML_SKIP_PYTORCH_TESTS", False),
+    reason="Skipping pytorch tests",
+)
+@pytest.mark.parametrize(
+    "fisher_algorithm",
+    [
+        FisherInverseFastSmallBlocks,
+        FisherInverseFastBlock,
+    ],
+)
+@pytest.mark.parametrize(
+    "devices",
+    [
+        None,
+        ["cuda:0"],
```

Comment:
causing CPU only test failure

Reply:
That explains it, thanks for catching that. I updated it so that pruning_mfac tests are all cpu based and the mfac_inverse tests that use gpu are skipped if a device is not found. The run pathways for cpu and gpu based runs diverge a good bit, which means that testing on a local machine with a GPU will test all the pathways in MFAC, but the tests on GHA unfortunately will not. Wanted to note this so that everyone is aware.